### PR TITLE
added load resource pack with offset feature

### PR DESCRIPTION
**Pull request by Arivval**
Added an optional int offset field for ```load_resource_pack()```, ```add_packs()``` and ```try_open_pack()``` in order to support .PCK file loading with offset. Need to update corresponding Godot Docs when these changes are merged.
With this PR, developers can load PCK with offsets using this syntax: 
Screen Shot 2020-07-13 at 4 28 00 PM
Screen Shot 2020-07-13 at 4 27 30 PM

### CODE REVIEW

File: core/io/file_access_pack.cpp
Comment (Arivval) [inline core/io/file_access_pack.cpp]: We suppose that load with offset feature won't be used together with [self contained exe feature](https://github.com/godotengine/godot/pull/24086). This allows us to only add one extra argument (but not size), giving us a cleaner interface.

File: core/io/file_access_zip.cpp
Comment (Arivval) [inline core/io/file_access_zip.cpp:150]: We also need to modify function signature here because it also implements the pure virtual ```try_open_pack()``` function.

File: core/io/file_access_zip.h
Comment (Arivval) [inline core/io/file_access_zip.h:72]: We also need to modify function signature here because it also implements the pure virtual ```try_open_pack()``` function.

Comment (Arivval) [reply core/io/file_access_zip.h:72]: Loading [ZIP](https://docs.godotengine.org/en/stable/getting_started/workflow/export/exporting_pcks.html) package is not within the scope of this PR (Install-time packs will use the format of .PCK package since they are already zipped inside of .apk when downloaded). Adding such feature will also require low level modification to Zlib. [here](https://github.com/godotengine/godot/blob/00949f0c5fcc6a4f8382a4a97d5591fd9ec380f8/core/io/file_access_zip.cpp#L172) and [here](https://github.com/godotengine/godot/blob/24e1039eb6fe32115e8d1a62a84965e9be19a2ed/thirdparty/minizip/unzip.h#L184)

File: core/io/file_access_pack.cpp
Comment (jamcohen) [inline core/io/file_access_pack.cpp]: Do we need to do a range check of any kind to make sure the offset is within the file?

File: core/io/file_access_zip.cpp
Comment (jamcohen) [inline core/io/file_access_zip.cpp:155]: Can we return false from this method if offset != 0?
That seems safer than silently ignoring it.

**Review by jamcohen — CHANGES_REQUESTED** [review]

File: core/io/file_access_pack.cpp
Comment (Arivval) [reply core/io/file_access_pack.cpp]: I double-checked ```seek()```'s implementation: [Android](https://github.com/godotengine/godot/blob/00949f0c5fcc6a4f8382a4a97d5591fd9ec380f8/platform/android/file_access_android.cpp#L75) [Unix](https://github.com/godotengine/godot/blob/00949f0c5fcc6a4f8382a4a97d5591fd9ec380f8/drivers/unix/file_access_unix.cpp#L187) [Windows](https://github.com/godotengine/godot/blob/00949f0c5fcc6a4f8382a4a97d5591fd9ec380f8/drivers/windows/file_access_windows.cpp#L192)
This EOF situation will be handled.

File: core/io/file_access_zip.cpp
Comment (Arivval) [reply core/io/file_access_zip.cpp:155]: Modified.

File: core/io/file_access_pack.cpp
Comment (thanwah) [reply core/io/file_access_pack.cpp]: This seems fair, but maybe add a note about this restriction as a code comment.

File: core/io/file_access_zip.cpp
Comment (thanwah) [inline core/io/file_access_zip.cpp]: Please set up and run clang-format as described in Aman's doc. (This will fix indents.)

File: core/io/file_access_zip.cpp
Comment (thanwah) [inline core/io/file_access_zip.cpp]: If this is a developer usage error, should this print an error message or anything?

**Review by thanwah — CHANGES_REQUESTED** [review]

File: core/io/file_access_zip.cpp
Comment (Arivval) [reply core/io/file_access_zip.cpp]: Formatted using clang-format

File: core/io/file_access_pack.cpp
Comment (Arivval) [reply core/io/file_access_pack.cpp]: Added comment.

File: core/io/file_access_pack.cpp
Comment (Arivval) [inline core/io/file_access_pack.cpp:138]: Added comment.

File: core/io/file_access_zip.cpp
Comment (Arivval) [reply core/io/file_access_zip.cpp]: Added error logging using [ERR_FAIL_COND_V_MSG()](https://github.com/godotengine/godot/blob/7e38e8fdb0c05f0216e4322806315a025dc05b6f/core/error_macros.h#L383).
If load zip file with offset is called, the debugger will log an error:
Screen Shot 2020-07-14 at 2 16 33 PM

Comment (Arivval) [reply core/io/file_access_zip.cpp]: Updated the error message to: Load with offset feature not supported for Zip files.

Comment (thanwah) [reply core/io/file_access_zip.cpp]: Thanks!

**Review by thanwah — APPROVED** [review]
LGTM, but please wait for Jake's approval.

File: core/io/file_access_zip.cpp
Comment (jamcohen) [reply core/io/file_access_zip.cpp]: It looks like [file_access_pack.cpp](https://github.com/godotengine/godot/blob/c2151e18135817c9f926a5a00341016ac77301d4/core/io/file_access_pack.cpp#L39) calls try_open_pack for each of it's sources. Since [sources is a vector of PackSources](https://github.com/godotengine/godot/blob/c2151e18135817c9f926a5a00341016ac77301d4/core/io/file_access_pack.h#L94), it might be possible to call try_open_pack with an offset on a zip pack without meaning to. Would this error prevent a user with zip packs from using PAD?

**Review by jamcohen — APPROVED** [review]
LGTM!

File: core/io/file_access_zip.cpp
Comment (jamcohen) [reply core/io/file_access_zip.cpp]: As discussed offline we'll keep the error as is for now and see what the Godot maintainers think.

**Pull request merged**
